pkg/utils/builders.go

```diff
@@ -22,12 +22,45 @@ type Builder struct {
 	errs            []error
 }
 
+// BuildContainerSpec builds a Container with following properties
+func BuildContainerSpec(perExperiment ExperimentDetails, engineDetails EngineDetails, envVar []corev1.EnvVar, volumeMounts []corev1.VolumeMount) *container.Builder {
+	containerSpec := container.NewBuilder().
+		WithName(perExperiment.JobName).
+		WithImage(perExperiment.ExpImage).
+		WithCommandNew([]string{"/bin/bash"}).
+		WithArgumentsNew(perExperiment.ExpArgs).
+		WithImagePullPolicy("Always").
+		//WithVolumeMountsNew(volumeMounts).
+		WithEnvsNew(envVar)
+
+	if volumeMounts != nil {
+		log.Info("Building Container with VolumeMounts")
+		log.Info(volumeMounts)
+		containerSpec.WithVolumeMountsNew(volumeMounts)
+	}
+
+	return containerSpec
+
+}
+
 // DeployJob the Job using all the details gathered
 func DeployJob(perExperiment ExperimentDetails, engineDetails EngineDetails, envVar []corev1.EnvVar, volumeMounts []corev1.VolumeMount, volumeBuilders []*volume.Builder) error {
 
 	// Will build a PodSpecTemplate
 	// For creating the spec.template of the Job
-	pod := BuildPodTemplateSpec(perExperiment, engineDetails, envVar, volumeMounts, volumeBuilders)
+	pod := BuildPodTemplateSpec(perExperiment, engineDetails, envVar)
+
+	// Add VolumeBuilders, if exists
+	if volumeBuilders != nil {
+		log.Info("Building Pod with VolumeBuilders")
+		log.Info(volumeBuilders)
+		pod.WithVolumeBuilders(volumeBuilders)
+	}
```

Comment:
I think lines 53-58 should be in `BuildPodTemplateSpec` as it s a part of building the podtemplate